src/Storages/MergeTree/MergeTreeDataPartInMemory.cpp

```diff
@@ -94,17 +119,59 @@ void MergeTreeDataPartInMemory::flushToDisk(const String & base_path, const Stri
     MergedBlockOutputStream out(new_data_part, metadata_snapshot, columns, indices, compression_codec);
     out.writePrefix();
     out.write(block);
+    const auto & projections = metadata_snapshot->getProjections();
+    for (const auto & [projection_name, projection] : projection_parts)
+    {
+        if (projections.has(projection_name))
+        {
+            String projection_destination_path = fs::path(destination_path) / projection_name / ".proj";
+            if (disk->exists(projection_destination_path))
+            {
+                throw Exception(
+                    ErrorCodes::DIRECTORY_ALREADY_EXISTS,
+                    "Could not flush projection part {}. Projection part in {} already exists",
+                    projection_name,
+                    fullPath(disk, projection_destination_path));
+            }
+
+            auto projection_part = asInMemoryPart(projection);
+            auto projection_type = storage.choosePartTypeOnDisk(projection_part->block.bytes(), rows_count);
+            MergeTreePartInfo projection_info("all", 0, 0, 0);
+            auto projection_data_part
+                = storage.createPart(projection_name, projection_type, projection_info, volume, projection_name + ".proj", parent_part);
+            projection_data_part->is_temp = false; // clean up will be done on parent part
+            projection_data_part->setColumns(projection->getColumns());
+
+            disk->createDirectories(projection_destination_path);
+            const auto & desc = projections.get(name);
+            auto projection_compression_codec = storage.getContext()->chooseCompressionCodec(0, 0);
+            auto projection_indices = MergeTreeIndexFactory::instance().getMany(desc.metadata->getSecondaryIndices());
+            MergedBlockOutputStream projection_out(
+                projection_data_part, desc.metadata, projection_part->columns, projection_indices, projection_compression_codec);
+            projection_out.writePrefix();
+            projection_out.write(projection_part->block);
+            projection_out.writeSuffixAndFinalizePart(projection_data_part);
+            new_data_part->addProjectionPart(projection_name, std::move(projection_data_part));
+        }
+    }
```

Comment:
Hm, I am confused how did it work before?
Does it mean `move partition` did not work if we had projections and in-memory parts?
If so, maybe we'd better have a separate bugfix